ygot/render_test.go

```diff
@@ -1580,11 +1580,17 @@ type testAnnotation struct {
 }
 
 func (t *testAnnotation) MarshalJSON() ([]byte, error) {
-	return json.Marshal(*t)
+	t2 := *t
+	t2.AnnotationFieldOne += t2.AnnotationFieldOne
+	return json.Marshal(t2)
 }
 
 func (t *testAnnotation) UnmarshalJSON(d []byte) error {
-	return json.Unmarshal(d, t)
+	if err := json.Unmarshal(d, t); err != nil {
+		return err
+	}
+	t.AnnotationFieldOne = t.AnnotationFieldOne[:len(t.AnnotationFieldOne)/2]
```

Comment:
Again, can you explain what is happening here please?

Reply:
As above.